src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+import {Position} from "./position"
+import {Probability} from "./probability"
+
+export class GameState{
+    public ShipPositions:Array<any>;
+    public MyShots:Array<any>;
+    public OpponentsShots:Array<any>;
+    public remainingShips:number[] = [2,3,3,4,5];  
+    // note that remainingShips always contains the shiplengths in ascending order
+    public getBoard():number[][]{
+        return this.board;
+    }
+    public static letterToNumber:any = {
+        "A":0, 
+        "B":1, 
+        "C":2, 
+        "D":3,
+        "E":4,
+        "F":5,
+        "G":6,
+        "H":7,
+        "I":8,
+        "J":9
+    };
+    public static numberToLetter:any = {
+        0:"A",
+        1:"B",
+        2:"C",
+        3:"D",
+        4:"E",
+        5:"F",
+        6:"G",
+        7:"H",
+        8:"I",
+        9:"J"
+    };
+
+    // 0 means unchecked, 1 means hit,2 means sunk, -1 means missed
+    public board:number[][] = [
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0]
+    ];
+
+    // reqBody is the request body, a json object
+    constructor(reqBody:any){
+        // A-J and 1-10
+        this.ShipPositions = (reqBody.ShipPositions)?(reqBody.ShipPositions):[];
+        this.MyShots = (reqBody.MyShots)?(reqBody.MyShots):[];
+        this.OpponentsShots = (reqBody.OpponentsShots)?(reqBody.OpponentsShots):[];
+        if (this.MyShots){
+            for (let i:number = 0; i<this.MyShots.length; i++){
+                let row:number = GameState.letterToNumber[this.MyShots[i].Position.Row];
+                let column:number = this.MyShots[i].Position.Column - 1
+                this.board[row][column] = (this.MyShots[i].WasHit)? 1 : -1;
+            }
+        }
+    }
+
+    public display():void{
+        console.log();
+        console.log(this.board);
+        console.log();
+    }
+
+    public isValidTarget(position:Position):boolean{
+        if (position.row >= 0){
+            if (position.row < 10){
+                if (position.column >= 0){
+                    if (position.column < 10){
+                        return (this.board[position.row][position.column]==0);
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public eliminateSunkenShips():void{
+        // first scans rows for sunken ships
+        for (let row:number =0; row<10; row++){
+            for (let column:number = 0; column<10; column++){
+                let startColumn:number = column;
+                if (this.board[row][column]==1){
```

Comment:
What is `this.board[row][column]==1`? I have no idea on a light reading!

This is where giving number[][] a specific class (see above comments) would be amazing. In that case, we could put on a method:

```
public isPreviousHit(tile: string): boolean = {
  let tileResult = this.getTile(tile);
  return tileResult.isHit;
}
```

Then this class only needs to know what a hit is, not how you've chosen to define a hit in you number[][] setup.